### DISCUSSION TITLE
Creating agile schedules in HA with OctpusEnergy addin.

### DISCUSSION THREAD

**Discussion by j0rdan** [BottlecapDave/HomeAssistant-OctopusEnergy#510] [Q&A] [answered]
Hi all

I'm very new here, and on Home Assistant in general, so please excuss my lack of knowledge.

I've got this addon installed and working, and also have a very basic automation going. Turn on underfloor heating when price drops below 0, turn off when price moves above.

Works, though I'm sure could be improved greatly :-)

My Question is abit more advanced than that, and looks more at the Target Rate sensor. I'm pretty sure I've got it added in correctly, to bring the cheapest slots covering a total 8 hours, inntermittent, between 4pm one day to the next (Agile user).

My question is, how I can use this data. As I said, I'm new to HA, so very wet behind the ears on all this. Currently, I can see the entity has an attribute of 'Target times' which shows the cheapest 16 30min blocks, but I'm at a loss of how to use this data.

I don't even know how to access this data to display it on a Dashboard card.

My main hope is to create a daily schedule to turn something on and off based on this data. Going forward this will be a Lay-Z pump, but for the meantime can use anything else as a test, or even just a log.

Any help would be appreciaited, if somebody could point me in the right direction of what methods I could use which would work with this addon would be amazing.

Regards
Jordan

**Comment by BottlecapDave** [comment, marked as answer]
The target rate sensor is a binary sensor, so you can use a normal state trigger on your automation looking at the target rate sensor. When state is 'on' then turn on your desired device, when state is 'off' then turn off your desired device.

**Reply by j0rdan** [reply]
Thanks Dave, litterally just figured it out, feel a bit silly it was so simple.

But like I said, completely new to HA and a little overwealmed by the option, so lots of learning to do.
Its very much a world of, its easy once you know how :-D

I'll keep reading, playing and learning.

If you know of any great learning resources, send em my way.  :-)
Once I've got this down I want to look at plotting this data for the coming 24 hour period, rather than just the past. Hopefully I can do that without the extras like Grafana.

Appriciate the addon by the way,

**Reply by BottlecapDave** [reply]
No problem, we all started somewhere :) and with home assistant, there's usually many different ways to accomplish the same task. It usually comes down to what makes sense to you :)
